pkg/advancedapi/advancedApi_test.go

```diff
@@ -613,3 +623,89 @@ func Test_can_create_seed(t *testing.T) {
 	_, err = advancedapi.CreateSeed(384)
 	assert.ErrorContains(t, err, "length must be 128 or 256")
 }
+
+func Test_can_not_reuse_subject_not_reusable_proof_for_delegation(t *testing.T) {
+	delegationFuncs := map[string]func(name string, controller string, proof string, revoked bool) register.RegisterDocumentOpts{
+		"controlDeleg": register.AddControlDelegation,
+		"authDeleg":    register.AddAuthenticationDelegation,
+	}
+	for name, delegationFunc := range delegationFuncs {
+		t.Run(name, func(t *testing.T) {
+
+			resolver := test.NewInMemoryResolver()
+			twin1ID, twin1Doc, err := advancedapi.CreateNewIdentityAndRegister(resolver, identity.Twin, test.ValidKeyPairPlop, "#ExistingId", false)
+			assert.NilError(t, err)
+			_, twin2Doc, err := advancedapi.CreateNewIdentityAndRegister(resolver, identity.Twin, test.ValidKeyPair3, "#ExistingId", false)
+			assert.NilError(t, err)
+			agentID, agentDoc, err := advancedapi.CreateNewIdentityAndRegister(resolver, identity.User, test.ValidKeyPairPlop2, "#ExistingId", false)
+			assert.NilError(t, err)
+
+			// create agent delegation proof (targeting a single delegating document: twin1)
+			agentIssuer, agentDelegProof, err := advancedapi.CreateDelegationProof(twin1ID.Issuer(), agentDoc, agentID.KeyPair())
+			assert.NilError(t, err)
+
+			// Add proof to twin1 document as a control delegation proof - the document is valid
+			regOpts := []register.RegisterDocumentOpts{
+				register.AddFromExistingDocument(twin1Doc),
+				delegationFunc("#controlDeleg1", agentIssuer.String(), agentDelegProof.Signature, false),
```

Comment:
substitute in the `name` param here & on line 660 maybe?

Reply:
The _name_ param here is only used for the test report display. 
The _delegation name_ can be anything starting with `#`, it has not impact on this test